File: python/数组/1427. 字符串的左右移.py
from typing import List
class Solution:
    def stringShift(self, s: str, shift: List[List[int]]) -> str:
        for direction,amount in shift:
            if amount == 0:
                continue
            if direction == 0:
                s = s[amount:]+s[:amount]
            else:
                s = s[len(s)-amount:]+s[:-amount]
        return s

S = Solution()
s = "joiazl"
shift = [[1,1],[1,6],[0,1],[1,3],[1,0],[0,3]]
print(S.stringShift(s,shift))

File: go/数组/1427. 字符串的左右移.go
package main

func stringShift(s string, shift [][]int) string {
	for _, t := range shift {
		direction, amount := t[0], t[1]
		if amount == 0 {
			continue
		}
		if direction == 0 {
			s = s[amount:] + s[:amount]
		} else {
			s = s[len(s)-amount:] + s[:len(s)-amount]
		}
	}
	return s
}
